ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v2/fileinput/clientlibs/site/js/fileinputview.js

```diff
@@ -86,6 +86,19 @@
             return this.element.querySelector(FileInputV2.selectors.attachButtonLabel);
         }
 
+        updateEnabled(enabled, state) {
```

Comment:
```
updateEnabled(enabled, state) {

    // Update data attribute for enabled state
    this.element.setAttribute(FormView.Constants.DATA_ATTRIBUTE_ENABLED, enabled);
    
    // Determine if the widget should be disabled
    const isDisabled = !enabled || state.readOnly;

    // Update the 'disabled' and ARIA 'disabled' attributes based on the enabled state
    if (isDisabled) {
        this.widget.setAttribute("disabled", "disabled");
        this.widget.setAttribute(FormView.Constants.ARIA_DISABLED, "true");
    } else {
        this.widget.removeAttribute("disabled");
        this.widget.removeAttribute(FormView.Constants.ARIA_DISABLED);
    }
}

```

Always try to create a constant inside the if condition to make the code more readable

Reply:
sure, implemented the same